plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -350,30 +350,35 @@ def _decide_drpms_to_download(self, metadata_files):
             _logger.debug('skipping DRPM sync')
             return set(), 0, 0
-        presto_file_handle = metadata_files.get_metadata_file_handle(presto.METADATA_FILE_NAME)
-        if presto_file_handle:
-            try:
-                package_info_generator = packages.package_list_generator(presto_file_handle,
-                                                                         presto.PACKAGE_TAG,
-                                                                         presto.process_package_element)
-                wanted = self._identify_wanted_versions(package_info_generator)
-                # check for the units that are already in the repo
-                not_found_in_the_repo = existing.check_repo(wanted.iterkeys(),
-                                                            self.sync_conduit.get_units)
-                # check for the units that are not in the repo, but exist on the server
-                # and associate them to the repo
-                to_download = existing.check_all_and_associate(not_found_in_the_repo,
-                                                               self.sync_conduit)
-                count = len(to_download)
-                size = 0
-                for unit in to_download:
-                    size += wanted[unit]
-            finally:
-                presto_file_handle.close()
-        else:
-            to_download = set()
-            count = 0
-            size = 0
+
+        to_download = set()
+        count = 0
+        size = 0
+
+        # multiple options for deltainfo files depending on the distribution
+        # so we have to go through all of them
+        for metadata_file_name in presto.METADATA_FILE_NAMES:
+            presto_file_handle = metadata_files.get_metadata_file_handle(metadata_file_name)
+            if presto_file_handle:
+                try:
+                    package_info_generator = packages.package_list_generator(
+                        presto_file_handle,
+                        presto.PACKAGE_TAG,
+                        presto.process_package_element)
+                    wanted = self._identify_wanted_versions(package_info_generator)
+                    # check for the units that are already in the repo
+                    not_found_in_the_repo = existing.check_repo(wanted.iterkeys(),
+                                                                self.sync_conduit.get_units)
+                    # check for the units that are not in the repo, but exist on the server
+                    # and associate them to the repo
+                    to_download = existing.check_all_and_associate(not_found_in_the_repo,
+                                                                   self.sync_conduit)
+                    count += len(to_download)
+                    for unit in to_download:
+                        size += wanted[unit]
+                finally:
+                    presto_file_handle.close()
+
```

Comment:
I guess no repo should have both files, right? If that's so, I guess it's ok, but I wonder if this loop should break once it finds one? I can't decide. Either way is good with me, just a thought.